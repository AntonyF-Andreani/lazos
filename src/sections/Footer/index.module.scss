.root {
  padding: 0 35px;
  width: auto;
  margin: 0 auto;

  @media (min-width: 576px) {
    max-width: 540px;
  }

  @media (min-width: 768px) {
    max-width: 720px;
  }

  @media (min-width: 992px) {
    max-width: 960px;
  }

  @media (min-width: 1200px) {
    max-width: 1140px;
  }

  @media (min-width: 1400px) {
    max-width: 1210px;
  }
}

.label {
  color: #777;
  text-decoration: none;
  background-color: transparent;
  line-height: 26px;
  font-size: 0.9em;

  & a {
    color: #777;
    text-decoration: none;
    outline: none;
    cursor: pointer;

    &:hover {
      color: #d71920;
      text-decoration: underline;
    }

    &:focus,
    &:active {
      color: #ccc;
    }
  }
}

.datos {
  padding-bottom: 1rem !important;
  padding-top: 1rem !important;
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
  box-sizing: border-box;
}

.wrapper {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
  box-sizing: border-box;
  display: block;

  @media (min-width: 768px) {
    flex: 0 0 50%;
    max-width: 50%;
  }

  @media (min-width: 992px) {
    flex: 0 0 25%;
    max-width: 25%;
  }
}

.logos {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;

  & :first-child {
    text-align: left;
  }

  & :last-child {
    text-align: right;
  }

  & span {
    flex: 0 0 50%;
    max-width: 50%;
  }
}

.icon {
  font-size: 1.3em;
  color: #d71920;
  margin-right: 10px;
}