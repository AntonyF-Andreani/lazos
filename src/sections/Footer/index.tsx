import { faDotCircle, faEnvelope } from "@fortawesome/pro-light-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Divider, Grid, List, ListItem, Typography } from "@mui/material";
import React from "react";
import { LazyLoadImage } from "react-lazy-load-image-component";

import styles from "./index.module.scss";

import type { Image } from "services/Cms/types";

export interface FooterProps {
  logoFundacion: Image;
  logoAndreani: Image;
}

const Footer = ({ logoFundacion, logoAndreani }: FooterProps) => {
  return (
    <div>
      <Divider variant="fullWidth" />
      <Grid container className={styles.root} direction="column">
        <Grid container item className={styles.datos}>
          <List className={styles.wrapper}>
            <ListItem>
              <Typography className={styles.label} color="secondary" component="span" variant="h6">
                <a href="https://forms.gle/qffkPw5A57u3DKK77" rel="noreferrer" target="_blank">
                  {"Formulario de inscripción y traslado"}
                </a>
              </Typography>
            </ListItem>
            <ListItem>
              <Typography className={styles.label} color="secondary" component="span" variant="h6">
                <a href="https://forms.gle/oWLgZmRJDo8reMBC9" rel="noreferrer" target="_blank">
                  {"Formulario de nuevo traslado"}
                </a>
              </Typography>
            </ListItem>
            <ListItem>
              <Typography className={styles.label} color="secondary" component="span" variant="h6">
                <a
                  href="https://andreanionline.com/hacer-un-envio"
                  rel="noreferrer"
                  target="_blank"
                >
                  {"Andreani Online"}
                </a>
              </Typography>
            </ListItem>
          </List>
          <List className={styles.wrapper}>
            <ListItem>
              <Typography className={styles.label} color="secondary" component="span" variant="h6">
                <a href="#" target="_blank" title="Descarga PDF">
                  {"Términos y condiciones"}
                </a>
              </Typography>
            </ListItem>
            <ListItem>
              <Typography className={styles.label} color="secondary" component="span" variant="h6">
                <a href="#" target="_blank" title="Simil a sitio web de Andreani?">
                  {"Politica de privacidad"}
                </a>
              </Typography>
            </ListItem>
          </List>
          <List className={styles.wrapper}>
            <ListItem>
              <FontAwesomeIcon className={styles.icon} icon={faDotCircle} />
              <Typography className={styles.label} color="secondary" component="span" variant="h6">
                {"Av. Don Pedro de Mendoza 1973, CABA"}
              </Typography>
            </ListItem>
            <ListItem>
              <FontAwesomeIcon className={styles.icon} icon={faEnvelope} />
              <Typography className={styles.label} color="secondary" component="span" variant="h6">
                <a href="mailto:lazos@fundacionandreani.org.ar" title="Falta tramitar">
                  {" lazos@fundacionandreani.org.ar"}
                </a>
              </Typography>
            </ListItem>
          </List>
          <List className={styles.wrapper}>
            <ListItem>
              <Typography className={styles.label} color="secondary" component="span" variant="h6">
                {"Copyright © 2021 Fundación Andreani. Todos los derechos reservados"}
              </Typography>
            </ListItem>
          </List>
        </Grid>
        <Divider variant="middle" />
        <Grid className={styles.logos}>
          <LazyLoadImage
            alt={logoFundacion.alt ? logoFundacion.alt : "Logo Fundacion"}
            className={styles.logoFundacion}
            effect="blur"
            src={logoFundacion.filename}
          />
          <LazyLoadImage
            alt={logoAndreani.alt ? logoAndreani.alt : "Logo Andreani"}
            className={styles.logoAndreani}
            effect="blur"
            src={logoAndreani.filename}
          />
        </Grid>
      </Grid>
    </div>
  );
};

export default Footer;
