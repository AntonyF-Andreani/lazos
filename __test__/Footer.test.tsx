import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import Footer, { FooterProps } from "../src/sections/Footer";

const defaultProps: FooterProps = {
  logoFundacion: {
    filename: `fundacion`,
    alt: "fundacion",
  },
  logoAndreani: {
    filename: `andreani`,
    alt: "andreani",
  },
};

describe("<Footer />", () => {
  it("should render without crash", () => {
    const { asFragment } = render(<Footer {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have the correct image", () => {
    const { getByAltText } = render(<Footer {...defaultProps} />);

    const logoFundacion = getByAltText("fundacion");
    const logoAndreani = getByAltText("andreani");

    expect(logoFundacion.getAttribute("src")).toBe(defaultProps.logoFundacion.filename);
    expect(logoAndreani.getAttribute("src")).toBe(defaultProps.logoAndreani.filename);
  });
});
