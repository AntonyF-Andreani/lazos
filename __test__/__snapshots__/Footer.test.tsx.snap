// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render without crash 1`] = `
<DocumentFragment>
  <div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column root css-1vpwcmr-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item datos css-1f064cs-MuiGrid-root"
      >
        <ul
          class="MuiList-root MuiList-padding wrapper css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          >
            <span
              class="MuiTypography-root MuiTypography-h6 label css-6zgmsr-MuiTypography-root"
            >
              <a
                href="https://forms.gle/qffkPw5A57u3DKK77"
                rel="noreferrer"
                target="_blank"
              >
                Formulario de inscripción y traslado
              </a>
            </span>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          >
            <span
              class="MuiTypography-root MuiTypography-h6 label css-6zgmsr-MuiTypography-root"
            >
              <a
                href="https://forms.gle/oWLgZmRJDo8reMBC9"
                rel="noreferrer"
                target="_blank"
              >
                Formulario de nuevo traslado
              </a>
            </span>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          >
            <span
              class="MuiTypography-root MuiTypography-h6 label css-6zgmsr-MuiTypography-root"
            >
              <a
                href="https://andreanionline.com/hacer-un-envio"
                rel="noreferrer"
                target="_blank"
              >
                Andreani Online
              </a>
            </span>
          </li>
        </ul>
        <ul
          class="MuiList-root MuiList-padding wrapper css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          >
            <span
              class="MuiTypography-root MuiTypography-h6 label css-6zgmsr-MuiTypography-root"
            >
              <a
                href="#"
                target="_blank"
                title="Descarga PDF"
              >
                Términos y condiciones
              </a>
            </span>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          >
            <span
              class="MuiTypography-root MuiTypography-h6 label css-6zgmsr-MuiTypography-root"
            >
              <a
                href="#"
                target="_blank"
                title="Simil a sitio web de Andreani?"
              >
                Politica de privacidad
              </a>
            </span>
          </li>
        </ul>
        <ul
          class="MuiList-root MuiList-padding wrapper css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-dot-circle fa-w-16 icon"
              data-icon="dot-circle"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 464c-118.663 0-216-96.055-216-216 0-118.663 96.055-216 216-216 118.663 0 216 96.055 216 216 0 118.663-96.055 216-216 216zm0-296c-44.183 0-80 35.817-80 80s35.817 80 80 80 80-35.817 80-80-35.817-80-80-80zm0 128c-26.467 0-48-21.533-48-48s21.533-48 48-48 48 21.533 48 48-21.533 48-48 48z"
                fill="currentColor"
              />
            </svg>
            <span
              class="MuiTypography-root MuiTypography-h6 label css-6zgmsr-MuiTypography-root"
            >
              Av. Don Pedro de Mendoza 1973, CABA
            </span>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-envelope fa-w-16 icon"
              data-icon="envelope"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 64H48C21.5 64 0 85.5 0 112v288c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V112c0-26.5-21.5-48-48-48zM48 96h416c8.8 0 16 7.2 16 16v41.4c-21.9 18.5-53.2 44-150.6 121.3-16.9 13.4-50.2 45.7-73.4 45.3-23.2.4-56.6-31.9-73.4-45.3C85.2 197.4 53.9 171.9 32 153.4V112c0-8.8 7.2-16 16-16zm416 320H48c-8.8 0-16-7.2-16-16V195c22.8 18.7 58.8 47.6 130.7 104.7 20.5 16.4 56.7 52.5 93.3 52.3 36.4.3 72.3-35.5 93.3-52.3 71.9-57.1 107.9-86 130.7-104.7v205c0 8.8-7.2 16-16 16z"
                fill="currentColor"
              />
            </svg>
            <span
              class="MuiTypography-root MuiTypography-h6 label css-6zgmsr-MuiTypography-root"
            >
              <a
                href="mailto:lazos@fundacionandreani.org.ar"
                title="Falta tramitar"
              >
                 lazos@fundacionandreani.org.ar
              </a>
            </span>
          </li>
        </ul>
        <ul
          class="MuiList-root MuiList-padding wrapper css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          >
            <span
              class="MuiTypography-root MuiTypography-h6 label css-6zgmsr-MuiTypography-root"
            >
              Copyright © 2021 Fundación Andreani. Todos los derechos reservados
            </span>
          </li>
        </ul>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
      />
      <div
        class="MuiGrid-root logos css-vj1n65-MuiGrid-root"
      >
        <span
          class=" lazy-load-image-background blur"
          style="display: inline-block;"
        >
          <img
            alt="fundacion"
            class="logoFundacion"
            src="fundacion"
          />
        </span>
        <span
          class=" lazy-load-image-background blur"
          style="display: inline-block;"
        >
          <img
            alt="andreani"
            class="logoAndreani"
            src="andreani"
          />
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
